app/models/credential_adoption_email_record.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class CredentialAdoptionEmailRecord < ApplicationRecord
+  validates :icn, presence: true
```

Comment:
(I guess this would be a schema change too) But I'd prob recommend a uniqueness constraint on ICN and email address

Reply:
Oh wait, unless you intend to have multiple entries based on different templates

Reply:
In that case, I'd recommend a uniqueness constraint on the intersection of ICN and email template